exchanges/binance/ratelimit.go

```diff
@@ -11,63 +11,188 @@ const (
 
 // RateLimit implements the request.Limiter interface
 type RateLimit struct {
-	GlobalRate *rate.Limiter
-	Orders     *rate.Limiter
+	SpotRate           *rate.Limiter
+	SpotOrdersRate     *rate.Limiter
+	UFuturesRate       *rate.Limiter
+	UFuturesOrdersRate *rate.Limiter
+	CFuturesRate       *rate.Limiter
+	CFuturesOrdersRate *rate.Limiter
 }
 
 // Limit executes rate limiting functionality for Binance
 func (r *RateLimit) Limit(f request.EndpointLimit) error {
 	var limiter *rate.Limiter
 	var tokens int
 	switch f {
-	case limitHistoricalTrades:
-		limiter, tokens = r.GlobalRate, 5
-	case limitOrderbookDepth500:
-		limiter, tokens = r.GlobalRate, 5
-	case limitOrderbookDepth1000:
-		limiter, tokens = r.GlobalRate, 10
-	case limitOrderbookDepth5000:
-		limiter, tokens = r.GlobalRate, 50
-	case limitOrderbookTickerAll:
-		limiter, tokens = r.GlobalRate, 2
-	case limitPriceChangeAll:
-		limiter, tokens = r.GlobalRate, 40
-	case limitSymbolPriceAll:
-		limiter, tokens = r.GlobalRate, 2
-	case limitOpenOrdersAll:
-		limiter, tokens = r.Orders, 40
-	case limitOrder:
-		limiter, tokens = r.Orders, 1
-	case limitOrdersAll:
-		limiter, tokens = r.Orders, 5
+	case spotDefaultRate:
+		limiter, tokens = r.SpotRate, 1
+	case spotOrderbookTickerAllRate,
+		spotSymbolPriceAllRate:
+		limiter, tokens = r.SpotRate, 2
+	case spotHistoricalTradesRate,
+		spotAccountInformationRate,
+		spotOrderbookDepth500Rate:
+		limiter, tokens = r.SpotRate, 5
+	case spotOrderbookDepth1000Rate:
+		limiter, tokens = r.SpotRate, 10
+	case spotPriceChangeAllRate:
+		limiter, tokens = r.SpotRate, 40
+	case spotOrderbookDepth5000Rate:
+		limiter, tokens = r.SpotRate, 50
+	case spotOrderRate:
+		limiter, tokens = r.SpotOrdersRate, 1
+	case spotOrdersAllRate:
+		limiter, tokens = r.SpotOrdersRate, 5
+	case spotOpenOrdersAllRate:
+		limiter, tokens = r.SpotOrdersRate, 40
+	case uFuturesDefaultRate,
+		uFuturesKline100Rate:
+		limiter, tokens = r.UFuturesRate, 1
+	case uFuturesOrderbook50Rate,
+		uFuturesKline500Rate,
+		uFuturesOrderbookTickerAllRate:
+		limiter, tokens = r.UFuturesRate, 2
+	case uFuturesOrderbook100Rate,
+		uFuturesKline1000Rate,
+		uFuturesAccountInformationRate:
+		limiter, tokens = r.UFuturesRate, 5
+	case uFuturesOrderbook500Rate,
+		uFuturesKlineMaxRate:
+		limiter, tokens = r.UFuturesRate, 10
+	case uFuturesOrderbook1000Rate,
+		uFuturesHistoricalTradesRate:
+		limiter, tokens = r.UFuturesRate, 20
+	case uFuturesTickerPriceHistoryRate:
+		limiter, tokens = r.UFuturesRate, 40
+	case uFuturesOrdersDefaultRate:
+		limiter, tokens = r.UFuturesOrdersRate, 1
+	case uFuturesBatchOrdersRate,
+		uFuturesGetAllOrdersRate:
+		limiter, tokens = r.UFuturesOrdersRate, 5
+	case uFuturesCountdownCancelRate:
+		limiter, tokens = r.UFuturesOrdersRate, 10
+	case uFuturesCurrencyForceOrdersRate,
+		uFuturesSymbolOrdersRate:
+		limiter, tokens = r.UFuturesOrdersRate, 20
+	case uFuturesIncomeHistoryRate:
+		limiter, tokens = r.UFuturesOrdersRate, 30
+	case uFuturesPairOrdersRate,
```

Comment:
For some reason from our order manager calling (*Binance).UAllAccountOpenOrders we are getting a 429 error. I cannot seem to find the reason why the weighting seems correct. This is the only issue I have encountered between USDT and COIN futures with all currency pairs enabled. Spot disabled.